pkg/codegen/internal/test/sdk_driver.go

```diff
@@ -4,23 +4,73 @@ import (
 	"os"
 	"os/exec"
 	"path/filepath"
+	"strings"
 	"testing"
 
-	"github.com/pulumi/pulumi/pkg/v3/codegen"
+	//"github.com/pulumi/pulumi/pkg/v3/codegen"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
-type ThenFunc func(t *testing.T, testDir string)
+type SdkTest struct {
+	// Short name of the directory, relative to `TestDir`.
+	Directory string
 
-type sdkTest struct {
-	Directory   string
 	Description string
-	Skip        codegen.StringSet
-	Then        map[string]ThenFunc
+
+	// Decides which programming languages to test the code
+	// generation for; defaults to all supported languages.
+	IncludeLanguage func(lang string) bool
+
+	// Hooks that run additional validations after the code has
+	// been generated. Hooks are usually language-specific and are
+	// included in this map by language.
+	HooksByLanguage map[string][]SdkTestHook
```

Comment:
This is perhaps too verbose but I was going for marking typecheck and test phases separately so they show up as separate steps in `go test -test.v`.